app/assets/javascripts/dashboard/dataset/team_assignment.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+import _ from "lodash";
+import * as React from "react";
+import { Select } from "antd";
+import { getEditableTeams } from "admin/admin_rest_api";
+import type { APITeamType } from "admin/api_flow_types";
+
+const { Option } = Select;
+
+type Props = {
+  allowedTeams: Array<APITeamType>,
+  onTeamsChange: Function,
+};
+
+type State = {
+  teams: Array<APITeamType>,
+};
+
+class TeamAssignment extends React.PureComponent<Props, State> {
```

Comment:
The class name + file name is not very informative. Perhaps `TeamSelectionComponent` is more obvious. Also take a look at the already existing `TeamSelectionView`. Perhaps this component can be used there as well.